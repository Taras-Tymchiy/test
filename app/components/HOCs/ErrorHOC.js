// @flow
import * as React from 'react';
import { View, StyleSheet, ActivityIndicator } from 'react-native';
import { TextRegular } from '../atoms/Text';
import { COLOR_TEXT_ERROR, COLOR_SCREEN_BG } from '../styles/constants';

type ErrorProps = {
  error?: Error
};

export default function ErrorHOC<Props: ErrorProps, Component: React.ComponentType<$Diff<Props, ErrorProps>>>(Comp: Component): React.ComponentType<Props> {
  return ({ error, ...props }: Props) => (
    <View>
      <Comp {...props}/>
      {error &&
        <View style={[ StyleSheet.absoluteFill, styles.overlay ]} >
          <TextRegular style={styles.errorText} >Oops</TextRegular>
          <TextRegular style={styles.errorText} >{error.message}</TextRegular>
        </View>}
    </View>
  );
};

const styles = StyleSheet.create({
  errorText: {
    color: COLOR_TEXT_ERROR
  },
  overlay: { 
    backgroundColor: COLOR_SCREEN_BG, 
    justifyContent: 'center',
    alignItems: 'center',
  }
});