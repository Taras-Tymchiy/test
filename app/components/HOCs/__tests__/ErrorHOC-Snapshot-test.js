import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import ErrorHOC from '../ErrorHOC';
import { View } from 'react-native';

const error = new Error('test');

it('renders a ErrorHOC wrapped View with error using Snapshots', () => {
  const WrappedView = ErrorHOC(View);
  expect(renderer.create(
    <WrappedView error={error} />
  )).toMatchSnapshot();
});
  
it('renders a ErrorHOC wrapped View without error using Snapshots', () => {
  const WrappedView = ErrorHOC(View);
  expect(renderer.create(
    <WrappedView />
  )).toMatchSnapshot();
});
