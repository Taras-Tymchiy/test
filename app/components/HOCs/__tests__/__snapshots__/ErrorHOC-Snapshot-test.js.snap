// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a ErrorHOC wrapped View with error using Snapshots 1`] = `
<View>
  <View />
  <View
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "alignItems": "center",
          "backgroundColor": "#EEE",
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#222",
            "fontSize": 16,
            "fontWeight": "normal",
          },
          Object {
            "color": "#F22",
          },
        ]
      }
    >
      Oops
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#222",
            "fontSize": 16,
            "fontWeight": "normal",
          },
          Object {
            "color": "#F22",
          },
        ]
      }
    >
      test
    </Text>
  </View>
</View>
`;

exports[`renders a ErrorHOC wrapped View without error using Snapshots 1`] = `
<View>
  <View />
</View>
`;
